File: solutions/javascript/bob/1/bob.js
export const hey = (message) => {
  const trimString = message.trim();

  if (isEmpty(trimString)) {
    return "Fine. Be that way!";
  }

  if (isQuestion(trimString) && isYelling(trimString)) {
    return "Calm down, I know what I'm doing!";
  }

  if (isQuestion(trimString)) {
    return "Sure.";
  }

  if (isYelling(trimString)) {
    return "Whoa, chill out!";
  }

  return "Whatever.";
};

const isEmpty = (input) => input === "";

const isQuestion = (input) => input.endsWith("?");

const isYelling = (input) =>
  input.toUpperCase() === input && input.toLowerCase() !== input;

File: solutions/csharp/bob/2/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement)
    {
        string trimString = statement.Trim();

        if (trimString =="")
        {
            return "Fine. Be that way!";
        }

        if ((trimString.EndsWith('?')) && (trimString.ToUpper() == trimString && trimString.ToLower() != trimString))
        {
            return "Calm down, I know what I'm doing!";
        }
        if (trimString.EndsWith('?'))
        {
            return "Sure.";
        }
        if (trimString.ToUpper() == trimString && trimString.ToLower() != trimString)
        {
            return "Whoa, chill out!";
        }


        else return "Whatever.";

    }
}